File: python/stdlib/number/const.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("Number Constant", "number.const", 1.0, {},
                                   {"result": "Number"},
                                   "Pass the number arg as output.", verbose, True, False)
        self.args = args

    def tick(self, value):
        return {"result": self.args}

File: java/src/main/java/de/michaelfuerst/graphprogramming/stdlib/string/Const.java
package de.michaelfuerst.graphprogramming.stdlib.string;

import de.michaelfuerst.graphprogramming.NodeSpecification;
import de.michaelfuerst.graphprogramming.stdlib.Node;

import java.util.HashMap;

/**
 * A string constant node.
 *
 * @author Michael Fürst
 * @version 1.0
 */
public class Const extends Node {
    private String constant;

    public HashMap<String, Object> tick(HashMap<String, Object> input) {
        HashMap<String, Object> result = new HashMap<String, Object>();

        result.put("result", constant);

        return result;
    }

    protected void initialize(HashMap<String, Object> args) {
        this.constant = (String)args.get("default");
    }

    protected boolean isInputNode() {
        return true;
    }

    public NodeSpecification getSpec() {
        NodeSpecification spec = new NodeSpecification("String Const", "stdlib.string.const", "Returns the arg on the output");
        spec.outputs.put("result", "String");
        return spec;
    }
}
